frontend/src/views/detail/hook/useDetailBook.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Book } from '../../../types';
+import { HTTP } from '../../../utils/http-methods';
+
+/**
+ * Custom Hook to handle the detail of a book
+ * @param {string} URL - The URL of the API
+ * @param {number} id - The id of the book
+ * @returns {Object} data - The data of the book
+ * @returns {boolean} isError - Indicates if there is an error or not
+ * @returns {boolean} isLoading - Indicates if the data is being loaded or not
+ * @returns {Function} deleteBookById - Function to delete a book by its id
+ */
+export const useDetailBook = (URL: string, id: number) => {
```

Comment:
this hook is similar to a "book client", try to create a vanilla-js book client which has the responsibility to handle http calls for the book(s), the endpoint URL and types. Then create simple hooks that use it.